pkg/blockstorage/awsefs/awsefs.go

```diff
@@ -61,13 +62,23 @@ func NewEFSProvider(config map[string]string) (blockstorage.Provider, error) {
 		return nil, errors.New("Account ID is empty")
 	}
 	accountID := *user.Account
-	efsCli := awsefs.New(s, aws.NewConfig().WithRegion(region))
-	backupCli := backup.New(s, aws.NewConfig().WithRegion(region))
+	creds := awsConfig.Credentials
```

Comment:
@hakanmemisoglu - this makes sense to me i.e. adding the ability to accept an AWS role as an input and using that if specified - this is the direction we want to head in for all AWS operations.

@julio-lopez - second pair of eyes please? Some context on why we're adding this. Instead of creating a "test user" that has the credentials required - we've created a "customer permissions" role - that we will allow all users to assume. The EFS tests are the first one to use that functionality

Reply:
It seems that we should encapsulate the support for AWS roles as part of creating either the config or the session for AWS ops. This could be used for block and potentially object store operations as well.
For now, it may be enough to factor out the role-related functionality into a separate function, and then we can figure out how to generalize it. Thoughts?